src/node/rpc/member_frontend.h

```diff
@@ -20,8 +22,147 @@
+      for (size_t i=0; i < claims_length; i++)
+      {
+        std::string claim_name(claims[i].name);
+        std::vector<uint8_t> claim_value(claims[i].value, claims[i].value + claims[i].value_size);
+        claims_map->emplace(std::move(claim_name), std::move(claim_value));
+      }
+      return OE_OK;
+    }
+
+    static int lua_verify_cert_and_get_claims(lua_State* l)
+    {
+      LOG_INFO_FMT("lua_verify_cert_and_get_claims");
+      nlohmann::json json = lua::check_get<nlohmann::json>(l, -1);
+      std::vector<uint8_t> cert_der = json;
+
+      std::map<std::string, std::vector<uint8_t>> claims;
+
+      oe_verifier_initialize();
+      oe_result_t res = oe_verify_attestation_certificate_with_evidence(
+        cert_der.data(),
+        cert_der.size(),
+        oe_verify_attestation_certificate_with_evidence_cb,
+        &claims);
+
+      if (res != OE_OK)
+      {
+        // TODO should this raise an exception?
+        throw std::runtime_error("certificate not valid");
+      }
+
+      lua_newtable(l);
+      const int table_idx = -2;
+
+      for (auto const& item : claims)
+      {
+        LOG_INFO_FMT("claim: {}", item.first);
+        // TODO not all claims should be hex-encoded
+        std::string val_hex = fmt::format("{:02x}", fmt::join(item.second, ""));
+        lua::push_raw(l, val_hex);
+        lua_setfield(l, table_idx, item.first.c_str());
+      }
+
+      return 1;
+    }
+#else
+    // Uses the old API: oe_verify_attestation_certificate.
+    // The quote is assumed to be an OE report (with oe_report_header_t header)
+    // stored at 1.2.840.113556.10.1.1 in the cert.
+
+    static oe_result_t oe_verify_attestation_certificate_cb(oe_identity_t* identity, void* arg)
+    {
+        std::memcpy(arg, identity, sizeof(oe_identity_t));
+        return OE_OK;
+    }
+
+    static int lua_verify_cert_and_get_claims(lua_State* l)
+    {
+      LOG_INFO_FMT("lua_verify_cert_and_get_claims");
+      nlohmann::json json = lua::check_get<nlohmann::json>(l, -1);
+      std::vector<uint8_t> cert_der = json;
+
+      oe_identity_t identity;
+      oe_result_t res = oe_verify_attestation_certificate(
+        cert_der.data(),
+        cert_der.size(),
+        oe_verify_attestation_certificate_cb,
+        &identity);
+
+      if (res != OE_OK)
+      {
+        throw std::runtime_error("certificate not valid");
+      }
+
+      lua_newtable(l);
+      const int table_idx = -2;
+
+      std::string mrsigner =
+        fmt::format("{:02x}", fmt::join(identity.signer_id, ""));
+      lua::push_raw(l, mrsigner);
+      lua_setfield(l, table_idx, "mrsigner");
+
+      std::string mrenclave =
+        fmt::format("{:02x}", fmt::join(identity.unique_id, ""));
+      lua::push_raw(l, mrenclave);
+      lua_setfield(l, table_idx, "mrenclave");
+
+      lua::push_raw(
+        l, static_cast<bool>(identity.attributes & OE_REPORT_ATTRIBUTES_DEBUG));
+      lua_setfield(l, table_idx, "is_debuggable");
```

Comment:
Is there a way we can generically extract and expose attributes?

Reply:
This is the old API, so no. In the new one (the one you commented on as well) it's already generic.